# STCC Classical Physics I: Newtonian Mechanics PHY-231

This Google Colab nb is a way for me to better understand my Physics 1 course I am taking at my local Community College. Hopefully it is helpful to you. Today we are going to use 3 basic kinematic equations and solve some applicable physics based problems.

## Kinematic Equations

Horizontal Motion:

 $v_0x=v_x , x = x_0 +v_x t$ 
 
 
 Vertical Motion: 
 
 $y = y_0 + \frac{1}{2}(v_0y + v_y) t$ 
 
 $v_y = v_0 - gt$
 
 $y=y_0 + v_0t - \frac{1}{2}gt^2$
 
 $v_y^2=v_0^2 -2g\Delta y$


The variables are:  
- final velocity, $v_f$
- initial velocity, $v_0$
- final velocity, $v_f$
- Change in position, $\Delta y$ $\equiv$  $(y - y_0)$
- time, $t$




## Calculating Projectile Motion: Tennis Player

A tennis player wins a match at Arthur Ashe stadium and hits a ball into the stands at 30 m/s and at an angle
45° above the horizontal (Figure 4.14). On its way down, the ball is caught by a spectator 10 m above the
point where the ball was hit. (a) Calculate the time it takes the tennis ball to reach the spectator. (b) What are the
magnitude and direction of the ball’s velocity at impact?
![alt text](https://github.com/ArtificialintelligenceDev/Physics_231/blob/master/Figure_4.14.PNG)

### Strategy to solve:
This example asks for the final velocity. Thus, we recombine the vertical and horizontal results to obtain $\overrightarrow{v}$ at final time t, determined in the first part of the example.

### Solution:
(a) While the ball is in the air, it rises and then falls to a final position 10.0 m higher than its starting altitude. We
can find the time for this by using Equation 4.22:

$y=y_0 + v_0t - \frac{1}{2}gt^2$

If we take the initial position $y_0$ to be zero, then the final position is y = 10 m. The initial vertical velocity is the
vertical component of the initial velocity:

$v_0 = v_0 sin \theta_0=(30.0m/s)sin45^{\circ}=21.1 m/s $



In [0]:
import math
v_0 = 30
y = v_0 * math.sin(45)
delta_Y = 10
g = 9.8

Substituting into Equation 4.22 for y gives us:

$(4.90 m/s^2)t^2 - (21.2 m/s)t + 10.0 m = 0$

In [5]:
# Solve the quadratic equation ax**2 + bx + c = 0

# import complex math module
import cmath

a = 4.9
b = -21.2
c = 10


# calculate the discriminant
d = (b**2) - (4*a*c)

# find two solutions
sol1 = (-b-cmath.sqrt(d))/(2*a)
sol2 = (-b+cmath.sqrt(d))/(2*a)

print('The solution are {0} and {1}'.format(sol1,sol2))

The solution are (0.5387960210504162+0j) and (3.7877345911944813+0j)


In [0]:
t = sol2

Using the quadratic formula yields t = 3.79 s and t = 0.54 s. Since the ball is at a height of 10 m at two times
during its trajectory—once on the way up and once on the way down—we take the longer solution for the time it
takes the ball to reach the spectator:

$t=3.79 s$

The time for projectile motion is determined completely by the vertical motion. Thus, any projectile that has an
initial vertical velocity of 21.2 m/s and lands 10.0 m below its starting altitude spends 3.79 s in the air.



(b) ) We can find the final horizontal and vertical velocities $v_x$ and $v_y$ and vy with the use of the result from (a). Then, we can combine them to find the magnitude of the total velocity vector $\overrightarrow{v}$ and the angle θ it makes with the horizontal. Since $v_x$ is constant, we can solve for it at any horizontal location. We choose the starting point because we know both the initial velocity and the initial angle. Therefore:
$v_x=v_0 cos(\theta_0)=(30 m/s) cos45^{\circ}= 15.75 m/s$

In [6]:
v_x = v_0 * math.cos(45)
v_x

15.759659664531892

The final vertical velocity is given by Equation 4.21:

$v_y = v_0 - gt$

Since $v_0$ was found in part (a) to be 21.2 m/s, we have:

$v_y = 21.2 m/s - 9.8 m/s^2 (3.79 s) = -7.11 m/s$

In [8]:
v_y = v_0 - g * t
v_y

(-7.1197989937059205+0j)

The magnitude of the final velocity $\overrightarrow{v}$ is :

$v = \sqrt{v_x^2 + v_y^2}= \sqrt{(21.2 m/s)^2 + (-15.9 m/s)^2}= 17.29 m/s $

In [11]:
# sqrt function doesn't allow complex numbers so we will use the approx. -7.11
v_y = -7.11
v = math.sqrt(v_x**2 + v_y**2)
v

17.289273343373154

The direction $ \theta_v is found using the inverse tangent:

$ \theta_v = tan^{-1}(\frac{v_y}{v_x}) = tan^{-1}(\frac{21.2}{-15.9})=$

In [14]:
import numpy as np
theta_v = np.arctan(v_y/v_x)
theta_v

-0.4238114129485152

## Significance
(a) As mentioned earlier, the time for projectile motion is determined completely by the vertical motion. Thus,
any projectile that has an initial vertical velocity of 21.2 m/s and lands 10.0 m below its starting altitude spends
3.79 s in the air. (b) The negative angle means the velocity is 53.1° below the horizontal at the point of impact.
This result is consistent with the fact that the ball is impacting at a point on the other side of the apex of the
trajectory and therefore has a negative y component of the velocity. The magnitude of the velocity is less than the
magnitude of the initial velocity we expect since it is impacting 10.0 m above the launch elevation.